### PR TITLE
Unify (IP, port) pairs in config using SocketAddr

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

Configuration currently contains several instances of (IP address, port) pairs. Rather than representing these pairs manually and individually, we should be using the SocketAddr type to encapsulate each pair into a single entity. This PR does this for AdmissionControlConfig and StorageConfig. It also removes the unused 'storage_node_debug_port'.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

A local run of 'cargo xtest' doesn't produce any issues.

## Related PRs

None.

### CODE REVIEW

File: config/src/config/storage_config.rs [hunk 1]
```diff
@@ -19,8 +19,9 @@ pub struct StorageConfig {
 impl Default for StorageConfig {
     fn default() -> StorageConfig {
         StorageConfig {
-            address: "localhost".to_string(),
-            port: 6184,
+            address: "127.0.0.1:6184"
+                .parse()
+                .expect("Unable to parse socket address"),
```
Comment (davidiw) [inline config/src/config/storage_config.rs]: I think in some of these cases and unwrap is good enough.

File: config/src/config/admission_control_config.rs [hunk 2]
```diff
@@ -18,8 +18,9 @@ pub struct AdmissionControlConfig {
 impl Default for AdmissionControlConfig {
     fn default() -> AdmissionControlConfig {
         AdmissionControlConfig {
-            address: "0.0.0.0".to_string(),
-            admission_control_service_port: 8000,
+            address: "0.0.0.0:8000"
```
Comment (davidiw) [inline config/src/config/admission_control_config.rs]: I think in some of these cases and unwrap is good enough.

File: storage/storage-client/src/lib.rs [hunk 3]
```diff
@@ -236,24 +237,24 @@ impl StorageRead for StorageReadServiceClient {
...
+    http_addr: String,
     client: Mutex<Option<StorageClient<tonic::transport::Channel>>>,
 }
 
 impl StorageWriteServiceClient {
-    /// Constructs a `StorageWriteServiceClient` with given host and port.
-    pub fn new(host: &str, port: u16) -> Self {
-        let addr = format!("http://{}:{}", host, port);
+    /// Constructs a `StorageWriteServiceClient` with given SocketAddr.
+    pub fn new(address: SocketAddr) -> Self {
```
Comment (davidiw) [inline storage/storage-client/src/lib.rs]: maybe pass by reference?

File: storage/storage-client/src/lib.rs [hunk 4]
```diff
@@ -41,24 +42,24 @@ use storage_proto::{
...
             client: Mutex::new(None),
-            addr,
+            http_addr,
         }
     }
 
     async fn client(&self) -> Result<StorageClient<tonic::transport::Channel>, tonic::Status> {
         if self.client.lock().unwrap().is_none() {
-            let client = StorageClient::connect(self.addr.clone())
+            let client = StorageClient::connect(self.http_addr.clone())
```
Comment (davidiw) [inline storage/storage-client/src/lib.rs]: maybe pass by reference? unless it needs the copy.

Comment (JoshLind) [reply storage/storage-client/src/lib.rs]: Looks like it needs the copy. The same goes for the connect() call in StorageWriteServiceClient.

File: config/src/config/admission_control_config.rs (hunk 2 above)
Comment (JoshLind) [reply config/src/config/admission_control_config.rs]: Done, thanks!

File: config/src/config/storage_config.rs (hunk 1 above)
Comment (JoshLind) [reply config/src/config/storage_config.rs]: Done, thanks!

File: storage/storage-client/src/lib.rs (hunk 3 above)
Comment (JoshLind) [reply storage/storage-client/src/lib.rs]: Done, thanks!

**Comment by JoshLind** [comment]
Awesome, thanks David! @bors-libra r=davidiw

**Pull request closed without merging**